main.js

```diff
@@ -160,4 +160,130 @@ const inventory = [
     originalStock: 10,
     sold: 8,
   },
-];
\ No newline at end of file
+];
+
+// Functionaliteit bouwen
+
+// Opdracht 1a: Gebruik een array-methode om een array te maken met alle tv-type namen. Log de uitkomst in de console.
+
+const typeOfTv = inventory.map((type) => {
+return type.type
+});
+
+//Opdracht 1b: Gebruik een array-methode om alle tv's te verzamelen (de hele objecten) die volledig uitverkocht zijn. Log de uitkomst in de console.
+
+const soldOud = inventory.filter((type) => {return type.originalStock === type.sold});
+// console.log(soldOud)
+
+
+
+// Opdracht 1c: Gebruik een array-methode om alle tv's te verzamelen (de hele objecten) die over AmbiLight beschikken. Log de uitkomst in de console.
+
+const ambiLightOptions = inventory.filter((lightOption) => {
+  return lightOption.options.ambiLight
+})
+// console.log(ambiLightOptions)
+
+//* **Opdracht 1d:** Schrijf een functie die alle tv's van laagste naar hoogste prijs sorteert. Log de uitkomst in de
+
+
+const sortPriceHiLo = inventory.map((price) => {
+  return price.price
+})
+
+
+inventory.sort((a,b) => a.price - b.price);
+
+
+
+
+function calculateMaxSold(object){
+  let soldTV = 0
+  for (let i = 0; i < object.length ; i++) {
+    soldTV += object[i].sold
+  }
+  return soldTV
+}
+
+
+// * **Opdracht 2b:** Zorg ervoor dat dit aantal _in het groen_ wordt weergegeven op de pagina.
+
+const publishTvPrice = document.getElementById("article-sold-tv")
+const articlePrice = document.createElement("h4")
+articlePrice.textContent = calculateMaxSold(inventory)
+articlePrice.setAttribute("class" , "color-sold-tv")
+publishTvPrice.appendChild(articlePrice)
+
+// * **Opdracht 2c:** Hoeveel tv's heeft Tech It Easy ingekocht? Schrijf een script dat dit berekent. Log de uitkomst in de
+
+
+function callMaxBought (object) {
+  let tvStock = 0
+  for (let i = 0; i < object.length; i++) {
+    tvStock += object[i].originalStock
+    tvStock++
+  }
+  return tvStock
+}
+
+const boughtTvHtml = document.getElementById("current-tv-bought")
```

Comment:
Eindig a.u.b. élke statement met een punt-komma. "**you should always use semicolons. Why? Because if you end up using a JavaScript compressor, all your code will be on one line, which will break your code**."